crmsh/sbd.py

```diff
@@ -174,34 +208,27 @@ def _update_configuration(self):
         utils.sysconfig_set(SYSCONFIG_SBD, **sbd_config_dict)
         bootstrap.csync2_update(SYSCONFIG_SBD)
 
-    def _determine_sbd_watchdog_timeout(self):
+    def _adjust_sbd_watchdog_timeout_with_diskless_and_qdevice(self):
         """
-        When using diskless SBD, determine value of SBD_WATCHDOG_TIMEOUT
+        When using diskless SBD with Qdevice, adjust value of sbd_watchdog_timeout
         """
         if not self.diskless_sbd:
             return
         # add sbd after qdevice started
         if utils.is_qdevice_configured() and utils.service_is_active("corosync-qdevice.service"):
             qdevice_sync_timeout = utils.get_qdevice_sync_timeout()
-            self._sbd_watchdog_timeout = qdevice_sync_timeout + 5
-            if self._is_s390 and self._sbd_watchdog_timeout < 15:
-                self._sbd_watchdog_timeout = 15
+            if self._sbd_watchdog_timeout <= qdevice_sync_timeout:
+                watchdog_timeout_with_qdevice = qdevice_sync_timeout + 5
+                bootstrap.warn("sbd_watchdog_timeout is set to {} for qdevice, it was {}".format(watchdog_timeout_with_qdevice, self._sbd_watchdog_timeout))
+                self._sbd_watchdog_timeout = watchdog_timeout_with_qdevice
             self._stonith_timeout = self.calculate_stonith_timeout(self._sbd_watchdog_timeout)
         # add sbd and qdevice together from beginning
         elif self._context.qdevice_inst:
-            self._sbd_watchdog_timeout = self.SBD_WATCHDOG_TIMEOUT_DEFAULT_WITH_QDEVICE
+            if self._sbd_watchdog_timeout < self.SBD_WATCHDOG_TIMEOUT_DEFAULT_WITH_QDEVICE:
+                bootstrap.warn("sbd_watchdog_timeout is set to {} for qdevice, it was {}".format(self.SBD_WATCHDOG_TIMEOUT_DEFAULT_WITH_QDEVICE, self._sbd_watchdog_timeout))
+                self._sbd_watchdog_timeout = self.SBD_WATCHDOG_TIMEOUT_DEFAULT_WITH_QDEVICE
```

Comment:
It occurred to me,  in case that `corosync.quorum.device.timeout` could be configured in profiles, of course it should check against that rather than the default.  But no urgency. Feel free to do the improvement now or some time in the future.

Reply:
Any suggest value for this timeout on different platform? If not clear yet, we could add it in the future